golem-common/src/precise_json.rs

```diff
@@ -0,0 +1,511 @@
+use golem_wasm_rpc::protobuf::type_annotated_value;
+use golem_wasm_rpc::protobuf::type_annotated_value::TypeAnnotatedValue;
+use serde::Deserialize;
+use serde::Serialize;
+use serde_json::Value as JsonValue;
+
+use thiserror::Error;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum PreciseJson {
```

Comment:
A couple of notes for this type
- Is there any difference from `wasm_rpc::Value`? I think there isn't?
- It misses resource handles 
- If we need a separate type (compared to `Value`) it still should be defined in `wasm_rpc` as all the other value types and conversions between them
- Why is there a custom decoder from json object - can't we use the default `serde_json` codecs (with some optional discriminator annotations if needed etc)?
- Where is the encoder

Reply:
There is an encoder now.
II can try to reuse Value in a separate PR